classes/01class/exercises/c01-aws02/obrientimothya/README.md

```diff
@@ -0,0 +1,201 @@
+# C01-AWS02
+
+## Commands Execution Output
+
+- Commands to create the S3 Bucket and to copy the file to the bucket:
+
+```
+AWS_ACCESS_KEY_ID=<HOWABOUTNO>
+AWS_SECRET_ACCESS_KEY=<NOTTODAYJIM>
+AWS_DEFAULT_REGION=us-east-1
+
+aws s3 mb s3://obrientimothyaaws02
+aws s3 cp ~/code/test.sh s3://obrientimothyaaws02/
+
+# output
+upload: ../test.sh to s3://obrientimothyaaws02/test.sh
+```
+
+- Commands to allow the EC2 instance to access the files in S3:
+
+```
+AWS_ACCESS_KEY_ID=<HOWABOUTNO>
+AWS_SECRET_ACCESS_KEY=<NOTTODAYJIM>
+
+aws iam create-policy --policy-name ReadAws02Bucket --policy-document file:///tmp/policy.json
+
+#output
+{
+    "Policy": {
+        "PolicyName": "ReadAws02Bucket",
+        "PolicyId": "ANPAWLZJXXALKPMMT4JOE",
+        "Arn": "arn:aws:iam::437637330966:policy/ReadAws02Bucket",
+        "Path": "/",
+        "DefaultVersionId": "v1",
+        "AttachmentCount": 0,
+        "PermissionsBoundaryUsageCount": 0,
+        "IsAttachable": true,
+        "CreateDate": "2020-06-23T06:47:56+00:00",
+        "UpdateDate": "2020-06-23T06:47:56+00:00"
+    }
+}
+
+cat <<EOF > /tmp/assumepolicy.json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+
+aws iam create-role --role-name EC2ReadAws02Bucket --assume-role-policy-document file:///tmp/assumepolicy.json
+
+# output
+{
+    "Role": {
+        "Path": "/",
+        "RoleName": "EC2ReadAws02Bucket",
+        "RoleId": "AROAWLZJXXALEX64FM355",
+        "Arn": "arn:aws:iam::437637330966:role/EC2ReadAws02Bucket",
+        "CreateDate": "2020-06-23T07:04:04+00:00",
+        "AssumeRolePolicyDocument": {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "Service": "ec2.amazonaws.com"
+                    },
+                    "Action": "sts:AssumeRole"
+                }
+            ]
+        }
+    }
+}
+
+
+cat <<EOF > /tmp/policy.json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetLifecycleConfiguration",
+                "s3:GetBucketTagging",
```

Comment:
do you need all that permission just to download a file from a bucket?

Reply:
Certainly not @caiocezart - I admit was my copy-paste (in a rush) form the policy creator.
I'll review it and I'll tighten it up 😊

Reply:
Tightened up S3 policy @caiocezart - please review. 👀